frontend/sumo-championship/src/contexts/UserContext.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { createContext, useState } from 'react';
+import { Role, postLogin } from '../api/login';
+import { loginInformation } from '../components/organisms/LoginForm';
+
+interface User{
+    isLogged: boolean,
+    name: string,
+    lastname: string,
+    email: string,
+    role: Role | null
+}
+
+const defaultUser: User = {
+    isLogged: false,
+    name: "",
+    lastname: "",
+    email: "",
+    role: null
+}
+
+interface UserContextType{
+    user: User,
+    signIn: (user: loginInformation) => Promise<void>,
+    signOut: () => void
+}
+
+export const userContext = createContext<UserContextType>({
+    user: defaultUser,
+    signIn: (user: loginInformation) => new Promise((res, rej) => {}),
+    signOut: () => {}
+});
+
+const UserContext = (props: { children: any; }) => {
```

Comment:
The context looks good, however I found [pretty interesting article](https://kentcdodds.com/blog/how-to-use-react-context-effectively) about how to use contexts in more optimal and efficient way. I used that in my previous project and it was perfect. You can find many templates/examples there

Reply:
Nice article, thanks! Took from that custom hook and undefined "trick" in typescript, but I think reducers are much over the top. Not really readable, simply seems like pure redux